cybergis_compute_client/CyberGISCompute.py

```diff
@@ -67,7 +68,96 @@ def __init__(self, url="cgjobsup.cigi.illinois.edu", port=443, protocol='HTTPS',
         self.job = None
         self.recentDownloadPath = None
 
-    def login(self, manualLogin=True, verbose=True):
+    def encrypt_token(self, token):
+        """
+        Encrypts the token using host variable.
+
+        Args:
+            token (str): User/Environment provided token.
+        """
+        self.jupyterhubApiToken = base64.b64encode((self.jupyterhubHost + '@' + token).encode('ascii')).decode('utf-8')
+
+    def get_jupyterhubHost(self, manualHost):
+        """
+        Gets the jupyterhub from the user.
+
+        Args:
+            manualHost (bool): set to True if host needs to be specified.
+        """
+        if (self.jupyterhubHost is None) or (not manualHost):
+            print("Please copy the JupyterHub url along with port. E.g http://127.0.0.1:8081")
+            self.jupyterhubHost = input('Enter your jupyterhubHost here: ')
+
+    def set_username(self):
+        """
+        Authenticates the token and saves the username.
+        """
+        res = self.client.request('GET', '/user', {"jupyterhubApiToken": self.jupyterhubApiToken})
+        self.username = res['username']
+
+    def save_token(self):
+        """
+        Writes token to json file.
+        """
+        with open('./cybergis_compute_user.json', 'w') as json_file:
+            json.dump({"token": self.jupyterhubApiToken}, json_file)
+
+    def login_token(self):
+        """
+        Saves username and token.
+        """
+        try:
+            self.set_username()
+            self.save_token()
+            return self.login()
```

Comment:
Make sure you document the return types of your functions.

Reply:
Technecially they are not returning anything so I added the types of the variables involved. Do you want me to do something like 
```
Returns:
    Call to login_token function.
```
?